### PR TITLE
Ability to Export currently installed packages

### PR DESCRIPTION
**Pull request by gep13**
- Using a simple PersistenceService, i.e. Open/Save Dialog
- Provide option to export the current list of installed packages to a packages.config file

### CODE REVIEW

**Comment by gep13** [comment]
@RichiCoder1 can I get you to case your eyes over this? Thanks!

File: Source/ChocolateyGui/ViewModels/Controls/LocalSourceControlViewModel.cs [hunk 1]
```diff
@@ -183,6 +197,37 @@ public bool ReceiveWeakEvent(Type managerType, object sender, EventArgs e)
...
+            try
+            {
+                var fileStream = this._persistenceService.SaveFile("*.config", "Config Files (.config)|*.config");
+
+                if (fileStream == null)
+                {
+                    return;
+                }
+
+                using (StreamWriter sw = new StreamWriter(fileStream, Encoding.ASCII))
```
Comment (RichiCoder1) [inline Source/ChocolateyGui/ViewModels/Controls/LocalSourceControlViewModel.cs]: Call me stickler, but I think we should be using actual Xml writing tools here.

File: Source/ChocolateyGui/ViewModels/Controls/LocalSourceControlViewModel.cs [hunk 2]
```diff
@@ -108,6 +117,11 @@ public bool CanUpdateAll()
             return this.Packages.Any(p => p.CanUpdate);
         }
 
+        public bool CanExportAll()
+        {
+            return this._exportAll;
```
Comment (RichiCoder1) [inline Source/ChocolateyGui/ViewModels/Controls/LocalSourceControlViewModel.cs]: If this is never false, you should be able to just leave of this from the command. Though, thinking about it now, we might want to set _exportAll = false inside ExportAll() so they don't accidentally try and export twice.

Comment (gep13) [inline Source/ChocolateyGui/ViewModels/Controls/LocalSourceControlViewModel.cs]: Yeah, that is probably a good shout :+1:

File: Source/ChocolateyGui/ViewModels/Controls/LocalSourceControlViewModel.cs (hunk 1 above)
Comment (gep13) [inline Source/ChocolateyGui/ViewModels/Controls/LocalSourceControlViewModel.cs]: Yip, this is just a PoC, to see what people's thoughts are.  Can switch to using Xml based classes for actual implementation.

**Comment by gep13** [comment]
@RichiCoder1 okay, made some changes based on your feedback, let me know what you think

**Comment by RichiCoder1** [comment]
Looks good.

**Comment by gep13** [comment]
What are your thoughts on something like this:

http://www.c-sharpcorner.com/UploadFile/mahesh/wpf-dropdownbutton-control/

I thought the export button could be overridable, i.e. default would be to create a packages.config file, but it could be selected to export to boxstarter script, or boxstarter url, or chocolatey package.

This would be a future enhancement, not for now, but just something to think about.

**Comment by RichiCoder1** [comment]
I'd be cool with that.

**Comment by gep13** [comment]
:+1:

**Pull request merged**